### DISCUSSION TITLE
FluxBC -> in the framework source code, not documented?

### DISCUSSION THREAD

**Discussion by smpeyres** [idaholab/moose#21375] [Q&A General] [answered]
I was browsing for a BC that I could use to set a flux at a boundary, agnostic of the exact physics inside the domain, when I came across FluxBC lying in the source code but isn't documented in https://mooseframework.inl.gov/moose/syntax/index.html 

Is this BC working and advisable for use? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think you should use a NeumannBC
I ll check this fluxBC but it s probably abusing its name

Guillaume

**Reply by GiudGiud** [reply]
There is no `FluxBC` with that name. But there are a couple of specific flux BCs, specific to the physics they are targetted at.

**Reply by smpeyres** [reply]
https://github.com/idaholab/moose/blob/master/framework/src/bcs/FluxBC.C

This is what I found, but it isn't documented.

**Reply by GiudGiud** [reply]
yeah use a neumannBC, this isnt for users

**Reply by GiudGiud** [reply]
for more info this is a base class. It s meant to be inherited from to make a specialized flux BC
